packages/block-editor/src/components/block-variation-picker/README.md

```diff
@@ -0,0 +1,54 @@
+# Block Variation Picker
+
+The `BlockVariationPicker` component allows to display for certain types of blocks their different variations, and to choose one of them.
+
+This component is currently used by the "Columns" block to display and choose the number and structure of columns. It is also used by the "Post Hierarchical Terms Block" block.
+
+![Columns block variations](https://make.wordpress.org/core/files/2020/09/colums-block-variations.png)
+
+## Table of contents
+
+1. [Development guidelines](#development-guidelines)
+2. [Related components](#related-components)
+
+## Development guidelines
+
+### Usage
+
+Renders the variations of a block.
+
+```jsx
+import { BlockVariationPicker } from '@wordpress/block-editor';
+
+const MyBlockVariationPicker = () => (
+	<BlockVariationPicker
+		variations={ variations }
+	/>
+);
+```
+
+### Props
+
+#### icon
+
+The icons for each variation of the block.
+
+#### label
+
+The label of each variation of the block.
+
+-   Type: `String`
+
+#### instructions
+
+The instructions to choose a block variation.
+
+-   Type: `String`
+
+#### variations
+
+The different variations of the block.
```

Comment:
I think we need to define what type this is, and what is expected.
It looks like it is an array of objects with name, icon, title, description properties.